src/lib.rs

```diff
@@ -1463,10 +1473,6 @@ impl EGraph {
         self.functions.values().map(|f| f.nodes.len()).sum()
     }
 
-    pub(crate) fn get_sort_from_value(&self, value: &Value) -> Option<&ArcSort> {
```

Comment:
Alternative proposal: change the type of `Value.tag` to `ArcSort`? (Can this be done everywhere actually?)

Reply:
`Symbol` is nicer in many cases because you can compare their equalities. Since sort names are unique, one can view the symbols as a unique reference to the actual sort. Changing Value.tag to ArcSort does not seem to introduce a significant benefit?

Reply:
If names define sort equality shouldn't we just
```rust
impl PartialEq for ArcSort {
    fn eq(&self, other: &ArcSort) -> bool {
        self.name() == other.name()
    }
}
```
The benefit is that we remove the extra lookup into the `type_info.sorts` map, making the code simpler for no cost (`name()` is faster than `get_sort_from_value()`).

Reply:
Conclusion: punt on this, it's fine for now